### DISCUSSION TITLE
Micro REPL - Open source MicroPython IDE for Android

### DISCUSSION THREAD

**Discussion by Ma7moud3ly** [micropython/micropython#11249] [Show and tell]
Hello Dears,
I'm very pleased to introduce the new open-source MicoPython IDE for Android "Micro REPL"
<img src="https://raw.githubusercontent.com/Ma7moud3ly/micro-repl/master/images/screens/feature_graphic.png"/>

Micro REPL is Thonny alike but for smartphones with many nice features   :D 
- Access to MicroPython REPL
- Files managers
- Code Editor

The app was tested on Android 12,11 devices, and Raspberry Pi Pico flashed with v1.19.1.uf2 firmware.
I really need your help to test and evaluate the app.. also all suggestions and contributions are welcomed.

https://play.google.com/store/apps/details?id=micro.repl.ma7moud3ly
https://github.com/Ma7moud3ly/micro-repl

**Comment by ctimmer** [comment]
App looks great. Looking forward to testing this weekend. 

Curt

**Comment by Wind-stormger** [comment]
Hope to support more devices such as ESP32!

I test you app on SONY XQ-AT72, Android 12.

ESP32S3, micropython 1.19.1, REPL output from USB , ❌

ESP32S2, micropython 1.19.1, REPL output from USB , ❌

ESP32S2, micropython 1.19.1, REPL output from UART , ❌

RP2040, micropython 1.19.1, REPL output from USB , ✔

**Comment by IHOXOHI** [comment]
Works with a pyboard v1.1, d, pybstick... Feather M4 express seems to be a keyboard...
Goods outlooks...
But repl session isn't completely interactive... An input() command no works.... Most of script can't works...

**Reply by IHOXOHI** [reply]
finally, most of functions works fine now... (like a while 1!)
GOOD WORKS.

**Comment by paulp59pp** [comment]
wondering if this is working on a chromebook too.

**Comment by ruliyanto** [comment]
I've just downloaded it from play store and test with simple blink on RP2040 using Xiaomi pad 5 and bluetooth keyboard, the blink.py script worked but the app sometimes crashed.

![Screenshot_2023-12-25-16-27-17-922_micro.repl.ma7moud3ly.jpg](https://github.com/micropython/micropython/assets/31688264/760baf53-0444-4c7b-aeed-5473a8792a73)


![IMG_20231225_164147.jpg](https://github.com/micropython/micropython/assets/31688264/6ac66658-1f46-4539-91c9-962c51a2e8b1)

![Screenshot_2023-12-25-17-00-05-476_micro.repl.ma7moud3ly.jpg](https://github.com/micropython/micropython/assets/31688264/d3d68226-dad4-4e40-855f-9ed4ea98a434)

![Screenshot_2023-12-25-17-00-13-755_micro.repl.ma7moud3ly.jpg](https://github.com/micropython/micropython/assets/31688264/b88a60fa-a523-417a-aea0-628db969f230)

**Comment by DMaisterra** [comment]
Im programming pi pico with it, and is excellent.. but trying to access somewhere the sketch folder so can backup or upload without creating new file, i can get with it... 
Any help apreciated.

**Reply by IHOXOHI** [reply]
I don't have my android for the moment, but maybe you can try a "my file = open('my_file_name', 'w')", ..., in the repl session...

**Comment by yprasetyabudi** [comment]
Works with EEP32 C3 Super mini
